Mathlib/LinearAlgebra/TensorProduct/Subalgebra.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Submodule
+import Mathlib.RingTheory.TensorProduct.Basic
+import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
+
+/-!
+
+# Some results on tensor product of subalgebras
+
+## Linear maps induced by multiplication for subalgebras
+
+Let `R` be a commutative ring, `S` be a commutative `R`-algebra.
+Let `A` and `B` be `R`-subalgebras in `S` (`Subalgebra R S`). We define some linear maps
+induced by the multiplication in `S`, which are
+mainly used in the definition of linearly disjointness.
+
+- `Subalgebra.mulMap`: the natural `R`-algebra homomorphism `A ⊗[R] B →ₐ[R] S`
+  induced by the multiplication in `S`, whose image is `A ⊔ B` (`Subalgebra.mulMap_range`).
+
+- `Subalgebra.mulMap'`: the natural `R`-algebra homomorphism `A ⊗[R] B →ₗ[R] A ⊔ B`
+  induced by multiplication in `S`, which is surjective (`Subalgebra.mulMap'_surjective`).
+
+-/
+
+open scoped Classical TensorProduct
+
+open FiniteDimensional
+
+noncomputable section
+
+universe u v w
+
+namespace Subalgebra
+
+variable {R : Type u} {S : Type v}
+
+section Semiring
+
+variable [CommSemiring R] [Semiring S] [Algebra R S]
+
+variable (A : Subalgebra R S)
+
+/-- If `A` is a subalgebra of `S/R`, there is the natural `R`-algebra isomorphism between
+`i(R) ⊗[R] A` and `A` induced by multiplication in `S`, here `i : R → S` is the structure map.
+This generalizes `Algebra.TensorProduct.lid` as `i(R)` is not necessarily isomorphic to `R`. -/
+def lTensorBot : (⊥ : Subalgebra R S) ⊗[R] A ≃ₐ[R] A := by
```

Comment:
Should this not be called `lTensorOne` to match the submodule version? That matches the pattern of naming defs after their behavior not their type.

bors r-


bors d+

Reply:
Originally, `Submodule.lTensorOne` is about `1 : Submodule R S`. But according to discussion https://github.com/leanprover-community/mathlib4/pull/12498#discussion_r1586280439 it is changed to `⊥ : Subalgebra R S` (which as a submodule is equal to `1 : Submodule R S`) since it makes the statement and proof a little bit easier.

But for subalgebras, we don't have `1 : Subalgebra R S` but only `⊥ : Subalgebra R S`, so in my opinion it's better to talk about `lTensorBot` since it is for `⊥ : Subalgebra R S`. What's your opinion?

Reply:
Sure, will be added in a few minutes.